packages/calling-stateful-client/src/CallDeclarativeCommon.ts

```diff
@@ -149,7 +149,9 @@ class ProxyTeamsCaptions implements ProxyHandler<TeamsCaptions> {
       case 'startCaptions':
         return this._context.withAsyncErrorTeedToState(async (...args: Parameters<TeamsCaptions['startCaptions']>) => {
           this._context.setStartCaptionsInProgress(this._call.id, true);
-          const ret = await target.startCaptions(...args);
+          const ret = await target.startCaptions(...args).catch((e) => {
```

Comment:
This function uses inline async-await syntax. Instead of the `catch` chain, use a try-catch block
```
try {
  const ret = await target.startCaptions(...args);
  this._context.setSelectedSpokenLanguage(this._call.id, args[0]?.spokenLanguage ?? 'en-us');
  return ret;
} catch (e) {
  this._context.setStartCaptionsInProgress(this._call.id, false);
}
```

Reply:
This function is also wrapped in a withAsyncErrorTeedToState function. To prevent the error being caught and not propogating up to this parent wrapper, re-throw the error after handling ux behavior:
```ts
try {
  ...
} catch (e) {
  this._context.setStartCaptionsInProgress(this._call.id, false);
  throw e;
}
```

Reply:
Updated